### DISCUSSION TITLE
FOSS4G North America 2023 Impressions

### DISCUSSION THREAD

**Discussion by sjg-wdw** [maplibre/maplibre#344] [General]
FOSS4Gna is the North American version of the big Geospatial open source conference.  This was the first one since before the pandemic.  I was there presenting an overview of the Native Metal project.  I had a few thoughts and impressions I'd like to share, mostly on the business side of things.

MapLibre is well known and well liked.  I've heard nothing but good will for the project.

In terms of competitors, OpenLayers is one, but it's considered very slow in similar use cases.  I'd say Leaflet is really your main competitor in terms of usage, but 'competitor' isn't really the right word.  

Devs seem to graduate from Leaflet to MapLibre.  For example, the National Park Service is slowly upgrading many of its maps from one to the other.  Anything to help with that process might get you more adoption.

The Cloud Optimized data formats are huge this year.  We know about [GeoTiff](https://www.cogeo.org/), of course, but there are others joining that pantheon.  On the web side I'd look to solid, easy support for COG, FlatGeobuf, and PMTiles.  Perhaps also give Zarr a look, though it's very open ended.

I'm not as familiar with the MapLibre JS, but I know some of those do already work well.  For Native, I think we need a better way of reading formats like PMTiles and the others.

On the subject of OpenLayers, a feature I heard mentioned a bit was dynamic raster processing.  Basically just a filter on raster data to, for instance, recolor it.  I know people have experimented with that in MapLibre JS.  Might be worthwhile to expand on that.

3D was also big, turning up in presenters' lists of important areas in the coming year.  What they mean by 3D is always more ambiguous.  There's room for a sort of map dominated 3D alongside fully 3D interaction.  Think of looking at buildings from the side and top rather than going into them.

On the open source side, Cesium takes up a lot of the space.  There's a possibility of doing their 3D Tiles format and sharing some users from their community.

The way presenters used MapLibre JS was a bit different from some of the end-user focused apps and sites.  It was largely as a background for data display.  The map was there, but the data on top was of more interest.  Sometimes that data was displayed in MapLibre, sometimes it was Deck.gl or something similar.  PMTiles was popular and the community may sort of be edging away from static vector tiles.  It's worth keeping in the loop on that.

That usage pattern suggests making it easier for some of the sorts of data sources Deck.gl handles.  Not to replace Deck.gl.  Just look at some of the more basic "please overlay this data set" cases and see what can be simplified.

I think a MapLibre table would have been welcome there too, but I realize the logistics of that are non-trivial.

**Comment by wipfli** [comment]
Thanks for the notes!
